rubicon/objc/api.py

```diff
@@ -968,6 +1000,24 @@ def _new_from_class_statement(cls, name, bases, attrs, *, protocols):
             else:
                 class_register(ptr, attr_name)
 
+        # Add cleanup of ivars / properties to dealloc
+
+        old_dealloc = libobjc.class_getMethodImplementation(ptr, SEL("dealloc"))
+
+        def _new_delloc(objc_self, _cmd):
+
+            # Invoke dealloc callback of each property.
+            for attr_name, obj in attrs.items():
+                if isinstance(obj, objc_property):
+                    obj.dealloc_callback(objc_self, attr_name)
```

Comment:
For consistency with the "register" code above, I would check for the existence of the `dealloc_callback` attribute instead of a specific class. This would also make it easier to add `dealloc_callback`s on other classes in the future (although I can't think of any good use for this right now).

```suggestion
                try:
                    dealloc_callback = obj.dealloc_callback
                except AttributeError:
                    pass
                else:
                    dealloc_callback(objc_self, attr_name)
```

Reply:
Incidentally, is there any reason we are not using `hasattr` instead? It looks a bit more elegant and is also used in `ObjCProtocol.__new__()`.

Reply:
The point of using `try`/`catch AttributeError` is that it accesses the attribute only once and checks that it exists at the same time as getting its value. This matters mainly when you're dealing with multithreading and the attribute might be deleted in between the `hasattr` check and the value being read. That can't happen here though, so in this case there would be no problem with using the shorter `hasattr` version instead of a `try`/`catch`.